manager/integration/tests/test_rwx.py

```diff
@@ -0,0 +1,152 @@
+    4. Verify the volume named as the PV attached to longhorn-nfs-provisioner
+       `volume.ready == True`
+    5. Create a StatefulSet of 1 pod with VolumeClaimTemplate and
+       longhorn-nfs class in default namespace.
+    6. Wait for StatefulSet to come up healthy.
+    7. Create another statefulset with same pvc which got created with first
+       statefulset.
+    8. Wait for statefulset to come up healthy.
+    9. Start writing 800 MB data in first stateful set and start writing 500 MB
+       data in second stateful set.
+    10. Compute md5sum.
+    11. Check the data md5sum in longhorn-nfs-provisioner volume
+    """
+    pass
+
+
+@pytest.mark.skip(reason="TODO")
+def test_rwx_statefulset_scale_down_up():
+    """
+    Test Scaling up and down of pods attached to longhorn-nfs-provisioner
+    volume.
+
+    1. Create PodSecurityPolicy for longhorn-nfs-provisioner as sample yaml
+       https://raw.githubusercontent.com/longhorn/longhorn/master/examples
+       /rwx/01-security.yaml
+
+    2. Create serviceAccount, service, longhorn-nfs-provisioner deployment,
+       PVC and StorageClass as sample yaml https://raw.githubusercontent.com/
+       longhorn/longhorn/master/examples/rwx/02-longhorn-nfs-provisioner.yaml
+
+    3. Wait for longhorn-nfs-provisioner deployment to come up healthy.
+    4. Verify the volume named as the PV attached to longhorn-nfs-provisioner
+       `volume.ready == True`
+    5. Create a StatefulSet of 2 pods with VolumeClaimTemplate and
+       longhorn-nfs class in default namespace.
+    6. Wait for StatefulSet pods to come up healthy.
+    7. Write data and compute md5sum in both pods
+    8. Force delete pods.
+    9. Wait for pods to terminate.
+    10. Recreate the pods
+    11. Wait for new pods to come up.
+    12. Check the data md5sum in new pods
+    """
+    pass
+
+
+@pytest.mark.skip(reason="TODO")
+def test_rwx_offline_node_longhorn_nfs_provisioner():
+    """
+    Test moving of longhorn-nfs-provisioner pod from one node to another.
+
+    1. Create PodSecurityPolicy for longhorn-nfs-provisioner as sample yaml
+       https://raw.githubusercontent.com/longhorn/longhorn/master/examples
+       /rwx/01-security.yaml
+
+    2. Create serviceAccount, service, longhorn-nfs-provisioner deployment,
+       PVC and StorageClass as sample yaml https://raw.githubusercontent.com/
+       longhorn/longhorn/master/examples/rwx/02-longhorn-nfs-provisioner.yaml
+
+    3. Wait for longhorn-nfs-provisioner deployment to come up healthy.
+    4. Verify the volume named as the PV attached to longhorn-nfs-provisioner
+       `volume.ready == True`
+    5. Create a StatefulSet of 1 pod with VolumeClaimTemplate and
+       longhorn-nfs class in default namespace.
+    6. Wait for StatefulSet to come up healthy.
+    7. Write data and compute md5sum.
+    8. Shutdown the node where longhorn-nfs-provisioner is running.
+    9. Force delete the deployment in unknown state
```

Comment:
Why force delete the deployment, you should be able to add a liveness probe which would restart the pod.
That should be sufficient to be able to use the newly brought up nfs-provisioner.

Reply:
Make sense. Modified it.